### DISCUSSION TITLE
Problema sensori pir SONOFF e TUYA

### DISCUSSION THREAD

**Discussion by dante70** [indomus/forum#1332] [Hardware & Firmware]
Un saluto a tutti.
Ho da un paio di mesi installato Zigbeemqtt  usando come hub la chiavetta di Giotag il tutto esclusivamente per avere una semplicissima cosa e cioe' l'accensione automatica di 2 luci sotto le pensiline di casa quando viene rilevato movimento.
Ho acquistato 2 Sonoff SNZB 03 che regolarmrnte configurati hanno poi fatto da trigger all'automazione di accensione.
Nascono dopo poco tempo i problemi su uno di questi sensori che si e' '' impallato'' per piu' di una volta  rimanendo perennemente in 
stato di rilevamento movimento.
Durante le festivita' e' successo anche al secondo son off con il risultato che si e' scaricata la batteria.
Sicuro che fossero difettosi entrambi i sensori ho acquistato 2 pir Tuya Ts0202 e uno e' stato installato in un altro punto della pensilina per testarlo.
Morale della favola e' da oggi alle 13 che e' nello stato movimento rilevato e se non intervengo con il pulsantino di reset e riconfigurando da zero non risolvo.
Sono sfortunato io o questi sensori o il sistema ZB e' disgraziato di suo.(3 su 3 e anche di marche diverse)
SE qualcuno mi puo' consigliare altri sensori a batteria che magari ha avuto modo di testare
Grazie

**Comment by Scialla** [comment]
io, fossi in te, guarderei i log di z2m per cercare di capire...

Non ho PIR delle marche citate, ne ho un paio in RF e un paio in zigbee (aqara e ikea) che lavorano senza problemi.

Attaccati ai log di z2m e cerca di capire che è effettivamente il PIR che rimane attivo o se è HA che lo vede attivo.. insomma devi andare a tentoni per capire dove sta il problema

**Reply by snakuzzo** [reply]
Elimino il tuo log...devi usare il markdown!!!!!!

**Reply by dante70** [reply]
mi ritrovo con '' sensore pensilina davanti ''  con indirizzo '' 0x00124b0023b3ac36''   da ieri che non segnala il movimento.
Prima di fare il reset allego log dal quale io non capisco tanto.Mi sembra che il sensore sia vista da HA e non ci siano problemi.
se qualcuno ha vogli adi dare un'occhiata.

nel post precedente avevo usato markdown ma in anteprima  non vedevo nulla.
adesso ho riprovato e sembra a posto ma iin realta' si capisce poco in quanto e' tutto consecutivo.
se non va bene mi indicate come fare meglio
grazie

 ```
MQTT publish: topic 'zigbee2mqtt/bridge/response/networkmap', payload '{"data":{"routes":false,"type":"raw","value":{"links":[{"depth":1,"linkquality":68,"lqi":68,"relationship":1,"routes":[],"source":{"ieeeAddr":"0x00124b0023a96bd9","networkAddress":11200},"sourceIeeeAddr":"0x00124b0023a96bd9","sourceNwkAddr":11200,"target":{"ieeeAddr":"0x00124b00228131e7","networkAddress":0},"targetIeeeAddr":"0x00124b00228131e7"},{"depth":1,"linkquality":170,"lqi":170,"relationship":1,"routes":[],"source":{"ieeeAddr":"0x00124b0023b3ac36","networkAddress":13501},"sourceIeeeAddr":"0x00124b0023b3ac36","sourceNwkAddr":13501,"target":{"ieeeAddr":"0x00124b00228131e7","networkAddress":0},"targetIeeeAddr":"0x00124b00228131e7"},{"depth":1,"linkquality":7,"lqi":7,"relationship":1,"routes":[],"source":{"ieeeAddr":"0xa4c138d707ffc624","networkAddress":942},"sourceIeeeAddr":"0xa4c138d707ffc624","sourceNwkAddr":942,"target":{"ieeeAddr":"0x00124b00228131e7","networkAddress":0},"targetIeeeAddr":"0x00124b00228131e7"},{"depth":1,"linkquality":12,"lqi":12,"relationship":1,"routes":[],"source":{"ieeeAddr":"0xa4c138971c6c8213","networkAddress":45080},"sourceIeeeAddr":"0xa4c138971c6c8213","sourceNwkAddr":45080,"target":{"ieeeAddr":"0x00124b00228131e7","networkAddress":0},"targetIeeeAddr":"0x00124b00228131e7"}],"nodes":[{"definition":null,"failed":[],"friendlyName":"Coordinator","ieeeAddr":"0x00124b00228131e7","lastSeen":null,"networkAddress":0,"type":"Coordinator"},{"definition":{"description":"Motion sensor","model":"TS0202","supports":"occupancy, battery_low, tamper, battery, linkquality","vendor":"TuYa"},"friendlyName":"PIR PENSILINA LATO","ieeeAddr":"0xa4c138d707ffc624","lastSeen":1643553120708,"manufacturerName":"_TZ3000_msl6wxk9","modelID":"TS0202","networkAddress":942,"type":"EndDevice"},{"definition":{"description":"Motion sensor","model":"SNZB-03","supports":"occupancy, battery_low, tamper, battery, linkquality","vendor":"SONOFF"},"friendlyName":"PIR PENSILINA DAVANTI","ieeeAddr":"0x00124b0023b3ac36","lastSeen":1643054819580,"manufacturerName":"eWeLink","modelID":"MS01","networkAddress":13501,"type":"EndDevice"},{"definition":{"description":"Motion sensor","model":"SNZB-03","supports":"occupancy, battery_low, tamper, battery, linkquality","vendor":"SONOFF"},"friendlyName":"PIR PENSILINA DIETRO","ieeeAddr":"0x00124b0023a96bd9","lastSeen":1643555913098,"manufacturerName":"eWeLink","modelID":"MS01","networkAddress":11200,"type":"EndDevice"},{"definition":{"description":"Motion sensor","model":"TS0202","supports":"occupancy, battery_low, tamper, battery, linkquality","vendor":"TuYa"},"friendlyName":"PIR PROVA","ieeeAddr":"0xa4c138971c6c8213","lastSeen":1643553738002,"manufacturerName":"_TZ3000_msl6wxk9","modelID":"TS0202","networkAddress":45080,"type":"EndDevice"}]}},"status":"ok","transaction":"qt2gl-2"}'

```

**Reply by snakuzzo** [reply]
Questo è il topic per la mappa...è un casino e non serve a niente in questo caso.
Inoltre il fatto che tu veda l'entità su HA non vuol dire nulla. Dimenticati di HA per questa analisi.
Devi guardare solo i log di Zigbee2mqtt e vedere se viene loggato qualcosa nel momento in cui dovrebbe pubblicare qualcosa sul broker.

**Reply by dante70** [reply]
ho fatto ora una prova:
siamo passati sotto ai sensori delle pensiline  quello che funziona (pensilina dietro)ha pubblicato quello che si vede nel log,l'altro niente di niente.
E' questo il log che dovevo guardare?





```
Debug Received Zigbee message from 'PIR PENSILINA DIETRO', type 'commandStatusChangeNotification', cluster 'ssIasZone', data '{"extendedstatus":0,"zonestatus":1}' from endpoint 1 with groupID 0
Info MQTT publish: topic 'zigbee2mqtt/PIR PENSILINA DIETRO', payload '{"battery":81.5,"battery_low":false,"linkquality":75,"occupancy":true,"tamper":false,"voltage":2900}'
Debug Received Zigbee message from 'PIR PENSILINA DIETRO', type 'commandStatusChangeNotification', cluster 'ssIasZone', data '{"extendedstatus":0,"zonestatus":0}' from endpoint 1 with groupID 0
Info MQTT publish: topic 'zigbee2mqtt/PIR PENSILINA DIETRO', payload '{"battery":81.5,"battery_low":false,"linkquality":66,"occupancy":false,"tamper":false,"voltage":2900}'
```

**Reply by snakuzzo** [reply]
Quindi ok il primo sensore...ko l'altro.
Se su Z2M non trovi nulla, vuol dire che non c'è comunicazione tra il device ed il coordinator. Problema di batteria ? Distanza ? Tutte prove da fare

**Reply by dante70** [reply]
grazie,
adesso provo a cambiare la batteria con una nuova di zecca .La distanza e' di 3,5 m con muro da 40 cm ..... al massimo ricompro un altro sensore di una terza marca a scopo di prova.
Saro' sfortunato io .....vedo tanti utenti che usano zigbee.
Ho acquistato da poco valvole termostatiche ma mi son guardato bene da quelle zigbee

**Reply by snakuzzo** [reply]
e perché ? cosa avrebbero di sbagliato ? Zigbee è validissimo!
Il discorso copertura è soggettivo e può essere risolto sfruttando appunto la rete mesh. Se poi hai vincoli (come nel mio caso) per cui non hai modo di piazzare un router nel mezzo, allora il discorso cambia...però di base il range non è un problema.
Ad ogni modo prima di buttare soldi comprando altra roba, prendi quel sensore e fai le prove spostandolo vicino al coordinator

**Reply by dante70** [reply]
Pe quanto riguarda il sensore in oggetto funziona bene magari per 2 /3 settimane poi si blocca.
Non avevo mai cambiato batteria,ora l'ho fatto anche se la vecchia ha tensione di 3.1V e la nuova 3.25v.....dubito sia quello.
ora e' perfettamente attivo e stiamo a vedere.

Son poco esperto di zigbee ma se ho ben capito gli apparecchi che possono ripetere il segnale son solo quelli alimentati non a batteria e nel caso delle valvole termostatiche citate avrei dovuto come dici tu piazzare il coordinator in una zona centrale ma avendo la casa su 2 livelli secondo me serviva un ''rilancio''. Credo che l'unica pecca di quelle non zigbee possa essere la batteria
